fairseq/sequence_generator.py

```diff
@@ -438,6 +462,7 @@ def _generate(
 
                 # Choose the subset of the hypothesized constraints that will continue
                 self.search.prune_sentences(batch_idxs)
+                self.search.prune_negative_sentences(batch_idxs)
```

Comment:
This can be combined with `LexicallyConstrainedBeamSearch::prune_sentences`

Reply:
Done.